### PR TITLE
Login

### PR DESCRIPTION
**Pull request by TFDU**
Devloperforfun 2021_3_11_login_update
1, Finish XML layout file(Old UI still remains for testing)
2, Finish MVVM
3, Store token in UserManager

### CODE REVIEW

File: gradlew.bat [hunk 1]
```diff
@@ -1,84 +1,84 @@
-@if "%DEBUG%" == "" @echo off
```
Comment (jingweihu) [inline gradlew.bat:1]: why do we have to delete this?

File: app/src/main/java/com/laiteam/echowall/user/UserManager.kt [hunk 2]
```diff
@@ -18,6 +19,8 @@ class UserManager @Inject constructor(
     val isUserLoggedIn: Boolean
         get() = userComponent != null
 
+    var userToken: String? = ""
```
Comment (jingweihu) [inline app/src/main/java/com/laiteam/echowall/user/UserManager.kt:22]: if you make "' as default, varUserToken should be notnull.

File: app/src/main/res/layout/fragment_login.xml [hunk 3]
```diff
@@ -1,39 +1,145 @@
...
-        app:layout_constraintTop_toBottomOf="@id/submit"
-        android:text="Login Fragment To Register" />
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:textAlignment="center"
+        android:text="@string/register"
+        app:layout_constraintTop_toBottomOf="@+id/reset_password"
+        app:layout_constraintStart_toStartOf="parent" />
+
+    <ImageView
```
Comment (jingweihu) [inline app/src/main/res/layout/fragment_login.xml:111]: View should be fine

**Review by jingweihu — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/laiteam/echowall/user/UserManager.kt (hunk 2 above)
Comment (TFDU) [reply app/src/main/java/com/laiteam/echowall/user/UserManager.kt:22]: reset default value to null

File: app/src/main/res/layout/fragment_login.xml (hunk 3 above)
Comment (TFDU) [reply app/src/main/res/layout/fragment_login.xml:111]: changed to View

**Review by jingweihu — APPROVED** [review]

**Pull request merged**